### PR TITLE
Use GitHub CI to test project on LTS versions of Java

### PR DESCRIPTION
**Pull request by carldea**
Created a convenient GitHub CI workflow to test project. GitHub CI allows up to 4 concurrent runners.

### CODE REVIEW

**Comment by npgall** [comment]
Could you explain what this will do? Will it add a second CI integration?

CQEngine project is already using [travis-ci](https://travis-ci.org/github/npgall/cqengine) to test CQEngine with LTS versions of Java. So I'm not sure it makes sense to add another such integration. Unless you could explain what the benefits would be?

**Comment by carldea** [comment]
Niall, 

Sorry for the slow reply.  My response is below:
> Could you explain what this will do? Will it add a second CI integration?
> 

GitHub Actions has some advantages:

- Well Integrated (the user interface is another tab in your project on GitHub as opposed to another site)
- Github runners (up to four concurrent are free). Travis is phasing out the free service of runners for newer projects.
- The new underlying GitHub Actions for setting up Java uses an LTS version of OpenJDK without worries about Oracle's new commercial license requirements. You just want to built and test or deploy your artifacts. Also, OpenJDK versions from 6 to 17ea is supported.
- GitHub CI is much simpler using it's matrix of OS and Java version if you should choose to add or update to newer OS or Java versions respectively. 

> CQEngine project is already using [travis-ci](https://travis-ci.org/github/npgall/cqengine) to test CQEngine with LTS versions of Java. So I'm not sure it makes sense to add another such integration. Unless you could explain what the benefits would be?

There are literally hundreds of article on reasons, but I found this article as to why I switched:
https://www.jeffgeerling.com/blog/2020/travis-cis-new-pricing-plan-threw-wrench-my-open-source-works

In a nutshell if they are the same from a build perspective (cleaner, easier, in one place, and continue to be free) why not?
